invenio_records_rest/serializers/csv.py

```diff
@@ -0,0 +1,132 @@
+# -*- coding: utf-8 -*-
+#
+# This file is part of Invenio.
+# Copyright (C) 2016-2019 CERN.
+#
+# Invenio is free software; you can redistribute it and/or modify it
+# under the terms of the MIT License; see LICENSE file for more details.
+
+"""Marshmallow based DublinCore serializer for records."""
+
+from __future__ import absolute_import, print_function
+
+import csv
+
+from .base import PreprocessorMixin, SerializerMixinInterface
+from .marshmallow import MarshmallowMixin
+
+
+class Line(object):
+    """Object that implements an interface the csv writer accepts."""
+
+    def __init__(self):
+        """Initialize."""
+        self._line = None
+
+    def write(self, line):
+        """Write a line."""
+        self._line = line
+
+    def read(self):
+        """Read a line."""
+        return self._line
+
+
+class CSVSerializer(SerializerMixinInterface, MarshmallowMixin,
+                    PreprocessorMixin):
+    """CSV serializer for records.
+
+    Note: This serializer is not suitable for serializing large number of
+    records.
+    """
+
+    def __init__(self, *args, **kwargs):
+        """Initialize CSVSerializer.
+
+        :param csv_excluded_fields: list of complete paths of the fields that
```

Comment:
I must admit I find it more intuitive to specify what to include instead of what to exclude, but that's my personal opinion. I.e. either you include all if the user didn't specify anything, or you include just what the user specfied.